devops/jobs/AppWatcher.groovy

```diff
@@ -61,10 +61,11 @@ class AppWatcher {
 
         // for each application
         extraVars.get("APPS_TO_CONFIG").each { app_name, app_config ->
-            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/" + app_name) {
+            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/application-watchers/" + app_name + "-watcher") {
 
                 def app_repo = app_config.get("app_repo", '')
-                def app_repo_branch = app_config.get('app_repo_branch', 'master')
+                // if an application has no repository, the app_repo_branch should be blank so as not to set up webhooks
+                def app_repo_branch = !app_repo ? '' : app_config.get('app_repo_branch', 'master')
```

Comment:
@jibsheet Yeah, I realized polling would be pointless if there is no repository to poll... I don't think there are empty repos, so I'll remove that and just support webhooks or manual.